accepted/global-json-updates.md

```diff
@@ -0,0 +1,369 @@
+* User wants to always use a specific SDK patch version, and receive an error if it is not on the machine.
+  * The user cannot currently do this.
+* User wants to always run the latest patch of a particular SDK feature band.
+  * Due to Rule #2, the only way to do this is to remove the previous version.
+    * The impact of this is mitigated by work we did in .NET Core 3.0 Preview 3 to replace earlier patch versions.
+* User includes a feature that requires an SDK that corresponds to a SDK feature band or higher, and wants to run the best available tools.
+  * There is no mechanism to require at least a specific version, but otherwise use the latest, or the latest in a range.
+  * Upgrading to a newer SDK will provide no benefit to projects tied to an earlier version.
+* User wants to know what version of the SDK is being used.
+  * `global.json` is a subtle, hidden and easily forgotten marker – programmers may not understand the downsides of leaving a global.json that was suggested in a blog post in place – or they may forget it.
+* User rarely wants the risk of running the unsupported scenario of multiple SDK versions in a solution build.
+  * There is nothing that encourages users to place `global.json` in the repository root or the same level as the solution. Using different versions of the SDK for different projects in the solution is not a supported scenario.
+* Hosted CI, including Azure DevOps and Kudu, need to provide a small number of SDKs.
+  * If there is no patch of the major/minor/feature version (xyz in `x.y.znn`) present, the build fails, causing friction between user and host needs.
+
+## Proposal
+
+This proposal parallels [runtime roll-forward behavior as initially described](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/roll-forward-on-no-candidate-fx.md)} and [recent improvements](https://github.com/dotnet/designs/blob/master/accepted/runtime-binding.md). All of the features proposed here may not be included in Phase 1. It will include at least `usePreview` and `"rollforward": "disable"`
+
+The runtime options allow for three basic approaches: find nearest, find latest and find exact. 
+
+## Details
+
+* Runtime: From the [recent improvements](https://github.com/dotnet/designs/blob/master/accepted/runtime-binding.md) document" _In all cases except `disable` the highest available patch version is selected._"
+
+The `global.json` schema to offers support for all of the identified scenarios that can be fixed without altering how we find `global.json`:
+
+* `sdk`
+  * `version`
+    * The lowest SDK the resolver would select - any SDKs with lower versions are ignored.
+    * Identical to today - wildcards are not supported.
+    * Default: empty
+  * `usePreview`
+    * `true` or `false`
+    * Default: `true` to match today's behavior.
+  * `rollForward` (The `version` must be specified if `rollForward` is specified, except `latestMajor`.)
+    * There are four basic intents for roll-forward:
```

Comment:
Currently the implementation isn't case-insensitive, but if the runtime does it this way, the SDK should to (personally not a big fan of case-insensitive enums).

I'll fix the implementation PR.

Reply:
We should match the runtime here. Thanks @vitek-karas for the catch.

Reply:
Context: We allowed case-insensitive for framework roll forward because that setting can be specified in several different ways. For JSON it makes sense to use the same casing you have here. For command that is less obvious but might be OKish, but for environment variables it just didn't look right.